### PR TITLE
Updated and regenerated REST API

### LINKED ISSUE #135: Use `https://api.gravatar.com/v3/profiles/{id}` instead of `https://www.gravatar.com/{id}.json`
Generate the client code from the backend specs `gravatar-public-openapi.json`

### PR DESCRIPTION
**Pull request by hamorillo**
Closes #135 

### Description

**Note:** This is a huge PR. I'm so sorry about that. It's complicated to split it.

Updated model classes and code with the new REST API definition.

### Testing Steps

- Install the demo app and verify you can load the profile cards for different emails and everything looks as expected (Exactly as it was working before this PR).

https://github.com/Automattic/Gravatar-SDK-Android/assets/6974554/306c9dcb-5d30-445d-82d0-dad9faa95c42

### CODE REVIEW

File: demo-app/src/main/java/com/gravatar/demoapp/ui/DemoGravatarApp.kt [hunk 1]
```diff
@@ -252,7 +252,7 @@ private fun ProfileTab(modifier: Modifier = Modifier, onError: (String?, Throwab
                             profileState = UserProfileState.Loading
                             when (val result = profileService.fetch(Email(email))) {
                                 is Result.Success -> {
-                                    result.value.entry.firstOrNull()?.let {
```
Comment (maxme) [inline demo-app/src/main/java/com/gravatar/demoapp/ui/DemoGravatarApp.kt:256]: Finally we can drop the list :)

File: gravatar/src/main/java/com/gravatar/GravatarConstants.kt [hunk 2]
```diff
@@ -14,5 +14,5 @@ internal object GravatarConstants {
     const val GRAVATAR_API_BASE_URL = "https://api.gravatar.com/v1/"
 
     /** Gravatar base host */
-    const val GRAVATAR_BASE_URL = "https://gravatar.com"
+    const val GRAVATAR_BASE_URL = "https://api.gravatar.com/v3/"
```
Comment (maxme) [inline gravatar/src/main/java/com/gravatar/GravatarConstants.kt]: We should rename:
- `GRAVATAR_API_BASE_URL` to `GRAVATAR_API_BASE_URL_V1`
- `GRAVATAR_BASE_URL`  to `GRAVATAR_API_BASE_URL_V3` (double check it's only used in ProfileService)

And at the same time, clean up stuff in `GravatarSdkContainer` (rename `getRetrofit*Builder`).

Comment (hamorillo) [reply gravatar/src/main/java/com/gravatar/GravatarConstants.kt]: Great idea! Done in [4e7f5d6](https://github.com/Automattic/Gravatar-SDK-Android/pull/141/commits/4e7f5d622a0fceea3dad38528f719adb428e5c2f)

**Comment by maxme** [comment]
I smoke tested the changes on the recently opened endpoint and everything from `ProfileService` seems to work as expected.

**Comment by hamorillo** [comment]
The backend is working as expected so this PR can be reviewed and merged.

**Review by maxme — APPROVED** [review]
LGTM, verified_accounts and pronouns got fixed on the backend, we can go ahead and merge this 👍

**Pull request merged**